transport/rif_comms/node.py

```diff
@@ -1,51 +1,339 @@
-from typing import Any
+import json
+from typing import Any, List, Dict, NewType, Union
 
+import structlog
+from eth_utils import to_checksum_address, is_binary_address
+from gevent import Greenlet, killall, wait, spawn
+from gevent.event import Event
+from greenlet import GreenletExit
+
+from raiden.exceptions import InvalidAddress, UnknownAddress, UnknownTokenAddress
 from raiden.message_handler import MessageHandler
-from raiden.messages import Message
+from raiden.messages import (
+    Message,
+    SignedRetrieableMessage,
+    SignedMessage, Delivered,
+    Processed,
+    Ping,
+    Pong,
+    from_dict as message_from_dict
+)
 from raiden.raiden_service import RaidenService
+from raiden.transfer import views
+from raiden.transfer.identifiers import QueueIdentifier
+from raiden.transfer.mediated_transfer.events import CHANNEL_IDENTIFIER_GLOBAL_QUEUE
+from raiden.transfer.state import QueueIdsToQueues
+from raiden.utils import Address, pex
 from raiden.utils.runnable import Runnable
-from raiden.utils.typing import Address
+from transport.matrix.utils import _RetryQueue
+from transport.message import Message as TransportMessage
 from transport.node import Node as TransportNode
 from transport.rif_comms.client import RifCommsClient
+from transport.rif_comms.proto.api_pb2 import Notification, ChannelNewData
+
+_RoomID = NewType("_RoomID", str)
+log = structlog.get_logger(__name__)
 
 
 class RifCommsNode(TransportNode, Runnable):
+    log = log
 
     def __init__(self, address: Address, config: dict):
         TransportNode.__init__(self, address)
         Runnable.__init__(self)
         self._config = config
+        self._raiden_service: RaidenService = None
+
+        self._rif_comms_connect_stream: Notification = None
+        self._our_topic: Notification = None
+        self._our_topic_thread: Greenlet = None
```

Comment:
could you please explain the reason behind these `None`s?

Reply:
Definition of the variable basically, its ok to use just `self._rif_comms_connect_stream: Notification`  and dont assign it to any value? 

I think python3 supports that

Reply:
you cannot get away with instance variables type hints without explicit assignments (you can do so only with class variables and it would not be correct here).

i try to avoid `None` assignments as much as possible in constructor methods (i think they're untidy) but i'm not sure if you're only adding these for the type hints.

the problem in this case is that some of these variables are not set until a later point (like the `start` function) so the underlying issue is bigger than this one; it's a design issue.

it's not a blocker for me anyway, so if you just want the type hints i guess this is ok